pkg/util/gc_service_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"context"
+	"math"
+
+	"github.com/pingcap/check"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	pd "github.com/tikv/pd/client"
+)
+
+type gcServiceuite struct {
+	pdCli mockPdClientForServiceGCSafePoint
+}
+
+var _ = check.Suite(&gcServiceuite{
+	mockPdClientForServiceGCSafePoint{serviceSafePoint: make(map[string]uint64)},
+})
+
+func (s *gcServiceuite) TestCheckSafetyOfStartTs(c *check.C) {
+	ctx := context.Background()
+	s.pdCli.UpdateServiceGCSafePoint(ctx, "service1", 10, 60) //nolint:errcheck
+	err := CheckSafetyOfStartTs(ctx, s.pdCli, 50)
+	c.Assert(err.Error(), check.Equals, "startTs less than gcSafePoint: [tikv:9006]GC life time is shorter than transaction duration, transaction starts at 50, GC safe point is 60")
+	s.pdCli.UpdateServiceGCSafePoint(ctx, "service2", 10, 80) //nolint:errcheck
+	s.pdCli.UpdateServiceGCSafePoint(ctx, "service3", 10, 70) //nolint:errcheck
+	err = CheckSafetyOfStartTs(ctx, s.pdCli, 65)
+	c.Assert(err, check.IsNil)
+	c.Assert(s.pdCli.serviceSafePoint, check.DeepEquals, map[string]uint64{"service1": 60, "service2": 80, "service3": 70, "ticdc-changefeed-creating": 65})
+
+}
+
+type mockPdClientForServiceGCSafePoint struct {
+	baseMockPdClient
```

Comment:
```suggestion
	pd.Client
```

How about this? So that we do not need to impl all methods.